cpp/conversion.h

```diff
@@ -0,0 +1,706 @@
+#pragma once
+
+#include "rrtmgp_const.h"
+
+#include <netcdf.h>
+#include <stdexcept>
+
+#ifdef RRTMGP_ENABLE_KOKKOS
+
+/**
+ * Helper functions for the conversion to Kokkos
+ */
+
+namespace conv {
+
+// Copied from EKAT
+#define IMPL_THROW_RRT(condition, msg, exception_type)    \
+do {                                                      \
+  if ( ! (condition) ) {                                  \
+    std::stringstream _ss_;                               \
+    _ss_ << "\n FAIL:\n" << #condition  << "\n";          \
+    _ss_ << "\n" << msg;                                  \
+    throw exception_type(_ss_.str());                     \
+  }                                                       \
+} while(0)
+
+#define RRT_REQUIRE(condition, msg) IMPL_THROW_RRT(condition, msg, std::runtime_error)
+
+// Copied from EKAT
+template<typename T, int N>
+struct DataND {
+  using type = typename DataND<T,N-1>::type*;
```

Comment:
I can foresee some typos or other code errors that cause the instantiation of DataND with N<0...

If you added a `static_assert(N>=0, "SOME GOOD ERROR MSG");`, the compiler will stop immediately and give a clear error msg, rather than waste time on recursion and report some generic recursion limit error.

Reply:
I think we may just use the EKAT version of this once the conversion is done.